src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,67 @@
+package duke;
+
+import java.io.InputStream;
+import java.io.PrintStream;
+import java.util.Scanner;
+
+/**
+ * A UI build upon the chatting bot.
+ */
+public class Ui {
+
+    private Scanner in;
+    private PrintStream out;
+
+    /**
+     * The default constructor.
+     */
+    public Ui() {
+        this(System.in, System.out);
+    }
+
+    /**
+     * The constructor of this class.
+     */
+    public Ui(InputStream in, PrintStream out) {
+        this.in = new Scanner(in);
+        this.out = out;
+    }
+
+    /**
+     * The method that returns the user command.
+     *
+     * @return the user command
+     */
+    public String getUserCommand() {
```

Comment:
I personally consider this method not necessary, but separating the situatiton of empty input and wrong input is a good idea.